src/server/qtquick/private/wquickxdgshell.cpp

```diff
@@ -134,7 +134,15 @@ void WXdgSurfaceItem::onSurfaceCommit()
 {
     WSurfaceItem::onSurfaceCommit();
     if (auto popup = m_surface->handle()->toPopup()) {
-        setImplicitPosition(popup->getPosition() - contentItem()->position());
+        if (m_surface->parentLayerSurface()) {
```

Comment:
You can do it in WXdgSurface::getPopupPosition().
And don't check its parent is WLayerSurface.